src/main/screen_options.cpp

```diff
@@ -0,0 +1,1022 @@
+/*
+ * TheXTech - A platform game engine ported from old source code for VB6
+ *
+ * Copyright (c) 2009-2011 Andrew Spinks, original VB6 code
+ * Copyright (c) 2020-2024 Vitaly Novichkov <admin@wohlnet.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+
+#include <IniProcessor/ini_processing.h>
+#include <AppPath/app_path.h>
+#include <Logger/logger.h>
+#include <fmt_format_ne.h>
+
+#include "core/render.h"
+
+#include "global_constants.h"
+#include "global_dirs.h"
+#include "sound.h"
+#include "globals.h"
+#include "gfx.h"
+#include "graphics.h"
+#include "config.h"
+
+#include "cheat_code.h"
+#include "game_main.h"
+#include "main/game_info.h"
+#include "main/menu_main.h"
+#include "main/screen_options.h"
+#include "graphics/gfx_frame.h"
+#include "graphics/gfx_marquee.h"
+#include "main/menu_controls.h"
+#include "controls.h"
+
+namespace OptionsScreen
+{
+
+constexpr size_t SECTION_NONE = -1;
+size_t section_index = SECTION_NONE;
+std::vector<size_t> section_header_indexes;
+size_t next_section_index = 0;
+
+std::vector<size_t> visible_items;
+
+
+size_t cur_item = 0;
+size_t cur_scroll = 0;
+
+int8_t mouse_scroll_cooldown = -1;
+bool controls_ready = false;
+bool cur_item_changed = false;
+
+bool global_dirty = false;
+
+MarqueeState name_marquee;
+MarqueeState value_marquee;
+MarqueeState name_tooltip_marquee;
+MarqueeState value_tooltip_marquee;
+
+static std::string s_temp_string;
+
+// Config_t* const config_levels[] = {nullptr, nullptr, &g_gameInfo, &g_config_game_user, &g_config_episode_creator, &g_config_episode_user,
+//     &g_config_file_creator, nullptr, &g_config_cmdline, nullptr};
+
+inline void s_set_dirty()
+{
+    global_dirty = true;
+}
+
+inline void s_switch_section()
+{
+    if(section_index == SECTION_NONE || section_index >= g_options.m_options.size())
+        return;
+}
```

Comment:
Looks like this function makes nothing than checking the condition and it doesn't return anything :eyes:

What the true sense of this function right now?